src/nora/components/NoraTextAreaInput/NoraTextAreaInput.js

```diff
@@ -17,7 +18,76 @@ export const NoraTextAreaInput = ({
   onChange,
   ...rest
 }) => {
+  const {
+    formChangeHandler,
+    validator,
+    initialValue,
+    currentValue,
+    currentError,
+    setFieldTouched,
+  } = { ...rest }
+
   const classes = [styles.NoraTextInput, textClassName].join(' ')
+
+  // Set up validation hooks
+  const [getError, setError, , validate] = useErrorMessage(validator)
+
+  const [val, setVal] = useState(currentValue || initialValue || '')
+
+  const [touched, setTouched] = useState(initialValue ? true : false)
+
+  const [doValidation] = useInputValidation({
+    validate,
+    setError,
+    formChangeHandler,
+  })
+
+  const handleChange = (ev) => {
+    const val = ev.target.value
+    setVal(val)
+
+    // We call setTouched in onBlur, so can reliably call getter here
+    doValidation(val, touched)
+
+    if (onChange) onChange(ev)
+  }
+
+  const setAllTouched = () => {
+    // We set touched to change the react state, but it's async and
+    // processing still, so, we use a flag for doValidation
+    setTouched(true)
+    // Also tell the form we've been touched
+    if (setFieldTouched) {
+      setFieldTouched(true)
+    }
+  }
+
+  // Initial Value aka prefilled—are considered "touched", but must prevalidate
+  // which will in turn update the internal form state as to their validity
+  useEffect(() => {
+    if (!!formChangeHandler && initialValue) {
+      console.log('TextAreaInput useEffect -- calling doValidation')
+      doValidation(initialValue, '')
+    }
+  }, [])
+
+  const handleBlur = (ev) => {
+    setAllTouched()
+    doValidation(ev.target.value, true)
+    if (onBlur) onBlur(ev)
+  }
+
+  /**
+   * If consumer's used the brute force `value` prop, we prioritize that
+   * (this is similar to what we're doing in CheckboxInput for `checked`)
+   */
+  const resolvedValue = typeof value !== 'undefined' ? value : val
+
+  // TODO: this indicates a field level error on the NoraTextAreaInput
```

Comment:
I figure I'll just leave this here and we can refactor once we know if/how we want to handle inline errors for Nora